geomstats/geometry/matrices.py

```diff
@@ -36,13 +39,36 @@ def mul(a, b):
 
         Returns
         -------
-        c : array-like, shape=[n_samples, dim1, dim3]
+        eq : array-like boolean, shape=[n_samples]
         """
-        return gs.matmul(a, b)
+        is_vectorized = \
+            (gs.ndim(gs.array(a)) == 3) or (gs.ndim(gs.array(b)) == 3)
+        axes = (1, 2) if is_vectorized else (0, 1)
+        return gs.all(gs.isclose(a, b, atol=atol), axes)
```

Comment:
does `gs.allclose` work? 

https://docs.scipy.org/doc/numpy/reference/generated/numpy.allclose.html

Reply:
Not exactly, that's something @stefanheyder walked me through on the first day :)

`gs.all` would also test along the `n_samples` dimension and behave like: `[ mats ] -> True`

Reply:
BTW it's good to have a `[mats] -> [bool]` method so that we'll be able to add methods like: 
+ `keep_points = filter(belongs)` to filter out those outside 
+ `are_points = gs.all(belongs)` to end up with the same test as `gs.all([mats])`

Reply:
Yes, if the input is vectorized, the output should also be a vector of booleans. `np.allclose` checks whether the arrays as a whole are equal, it is not vectorized.

Reply:
Got it, yes the [mats] -> [bools] format would be better!